src/frame/endpoints/PolicyApiEndpoints.ts

```diff
@@ -0,0 +1,56 @@
+import {ErrorCodes} from "../../Constants";
+import * as ApiRequest from "../ApiRequest";
+import ApiState from "../ApiState";
+import Future from "futurejs";
+import {SDKError} from "../../shim";
+import {UserOrGroup, Policy} from "ironweb";
+
+export type UserOrGroupWithKey = UserOrGroup & {masterPublicKey: PublicKey<string>};
+
+interface PolicyApplyResponse {
+    usersAndGroups: UserOrGroupWithKey[];
+    invalidUsersAndGroups: UserOrGroup[];
+}
+
+/**
+ * Generate signature message from current user state
+ */
+function getSignatureHeader() {
+    const {segmentId, id} = ApiState.user();
+    return ApiRequest.getRequestSignature(segmentId, id, ApiState.signingKeys());
+}
+
+/**
+ * Create the policy apply request based on the `policy`.
+ * @param policy The policy to be applied.
+ */
+export function applyPolicy(policy: Policy) {
+    const classification = policy.classification !== undefined ? [`classification=${encodeURIComponent(policy.classification)}`] : [];
+    const category = policy.category !== undefined ? [`category=${encodeURIComponent(policy.category)}`] : [];
+    const dataSubject = policy.dataSubject !== undefined ? [`dataSubject=${encodeURIComponent(policy.dataSubject)}`] : [];
+    const id = policy.substituteId !== undefined ? [`id=${encodeURIComponent(policy.substituteId)}`] : [];
```

Comment:
It might be better to just do a more falsy check on these instead of explicitly checking `undefined`, e.g. `const category = policy.category ? ...`. We don't really want to send in a value if the user gives us an object with `null` or empty string values.

Reply:
The idea of falsey makes me :nauseated_face: so I'll just take your word for it.